src/approaches/oracle_approach.py

```diff
@@ -1001,3 +1010,183 @@ def _get_repeated_nextto_gt_nsrts() -> Set[NSRT]:
     nsrts.add(grasp_nsrt)
 
     return nsrts
+
+
+def _get_behavior_gt_nsrts() -> Set[NSRT]: # pragma: no cover
+    """Create ground truth nsrts for BehaviorEnv."""
```

Comment:
can you add a comment somewhere here explaining the type combination stuff, and how this stems from our lack of hierarchical type support?

Reply:
I understand why we need type combinations here, but unsure how hierarchical types would help? Could you maybe elaborate a bit on this

Reply:
yeah so if we had hierarchical types, we could effectively use the original PDDL operators that you had written, with just one pick operator, one placeontop operator, etc. the way that hierarchical types were used there was that every type inherited from a default object type, so you could have arguments to your PDDL operators that were of type `object`, and then you could use those operators with objects of type `book`, `carrot`, etc. since we don't have support for hierarchical types right now (actually, we might have some support, but let's just finish out this PR and then revisit!), we're hacking around this by one operator for every combination of types, like pick-book, pick-carrot, etc., where each operator has the specific subtypes in its parameters.